src/config.rs

```diff
@@ -0,0 +1,90 @@
+use crate::NodeFormatter;
+
+/// The Colors that can be displayed in the console
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Clone)]
+pub enum Color {
+    Black,
+    White,
+    Red,
```

Comment:
FYI soon-ish I plan to create a crate that defines ANSI terminal formatting but provides nothing else so that it is
- Lightweight
- High compatibility guarantees

The goal will be to allow it to be used for in public APIs while each crate may implement formatting in a variety of ways.  This is a problem that currently blocks customizing formatting with clap.

Reply:
Alright, Im gonna keep that in the back of my mind, this was just a quick way to add it without having to introduce a new dependency that might change in the future and since I'm still working on a lot more stuff before doing the next release I don't think it would be too hard to swap out in the Future/before the next release

Reply:
With the great motivator of wanting to procrastinate other work, I've gone ahead and created a prototype. 
- https://github.com/epage/anstyle
- https://docs.rs/anstyle ([still building](https://docs.rs/releases/queue) as of when I posted this)

The plan is for this to be a near-perpetual 1.0 once it hits that so its safe to put in public APIs.  I want to create some of the other planned crates around this to have more confidence before I release 1.0 though I suspect the current API is there as I plan to keep this pretty simple.  The focus was on compile times rather than developer productivity, so it keeps language features relatively simple (e.g. no macros to help write repetitive code)

Other planned crates
- Conversion between 4-bit, 8-bit, and 24-bit numbers to allow graceful degradation when a terminal doesn't support 24-bit color
- Composable streams that can
  - Degrade colors on the fly
  - Remove all styling
  - Adapt to the old windows color API
- High level stdout/stderr streams that detect their capabilities and adapt on the fly
- Conversion from ansi to roff to help with `clap_mangen`

Reply:
Would this crate also take care of the windows specific stuff or does it only do the actual formatting?

Reply:
`anstyle` is not intended to provide a full formatting API (like what you might see with `yansi`, `termcolor`, or `owo-colors`) though it does support rendering the codes so a crate like `termgraph` could use it for printing codes if it wants (I might just do that with clap in the future).

One of the future crates I mentioned will be another option for formatting that might be of interest:
> High level stdout/stderr streams that detect their capabilities and adapt on the fly

Right now, most formatting crates put a lot of the burden on format-site to know what stream you will output to when you might not always, especially if you are in a `Display::fmt`.  I think instead we should be making this decision when writing to the stdout/stderr stream.  We can do this by modifying the output on-the-fly.  This reduces performance when escape codes need to be processed (graceful degradation for the terminal, converting to the old windows API) but I suspect that performance loss in those cases is worth it.

This would allow a crate like this to not worry about how the colors will be used.  You just provide an API for specifying a color palette, including making it easy to disable all styling
- If the user is using this crate I plan to write, then everything will be done automatically
- If not, the user could use some of these lower level crates to adapt just termgraph to do what they want
- If all else fails, the user can disable colors with your color palette.